mne/decoding/time_gen.py

```diff
@@ -1375,3 +1446,71 @@ def _clean_times(self):
             self.y_pred_ = [y_pred[0] for y_pred in self.y_pred_]
         if hasattr(self, 'scores_'):
             self.scores_ = [score[0] for score in self.scores_]
+
+
+def _chunk_data(X, slices, gat, n_orig_epochs, test_epochs):
+    """Smart chunking to avoid memory overload.
+
+    The parallelization is performed across time samples. To avoid overheads,
+    the X data is splitted into large chunks of different time sizes. To
+    avoid duplicating the memory load to each job, we only pass the time
+    samples that are required by each job. The indices of the training times
+    must be adjusted accordingly.
+    """
+
+    # from object array to list
+    slices = [sl for sl in slices if len(sl)]
```

Comment:
is it another weird corner case? how do you get it? I don't understand how your code can run into so many weird corner cases. Since it figures in the diff of this PR, I think it should be fixed in this PR itself.